### PR TITLE
feat: [CDS-50313]: helm changes: Get Node information via API from CD Deployment Step for CI

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
helm changes: Get Node information via API from CD Deployment Step for CI

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50609)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java [hunk 1]
```diff
@@ -34,4 +37,12 @@ public HelmInstallCmdResponseNG(CommandExecutionStatus commandExecutionStatus, S
     this.releaseName = releaseName;
     this.helmVersion = helmVersion;
   }
+
+  public List<K8sPod> getPreviousK8sPodList() {
```
Comment (abosii-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java:41]: Do we need these changes for rollback?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java (hunk 1 above)
Comment (namantalaycha10) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java:41]: let me confirm with @yogesh-chauhan

Comment (namantalaycha10) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java:41]: yes I need to add changes for helm rollback as well

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 6 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50609)* status: 8 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained.

**Comment by namantalaycha10** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java (hunk 1 above)
Comment (namantalaycha10) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/helm/HelmInstallCmdResponseNG.java:43]: helm rollback I will do in different PR

**Comment by namantalaycha10** [comment]
trigger messagecheck

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 8 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50609)* status: 9 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]

**Comment by namantalaycha10** [comment]
trigger review

**Pull request merged**